### PR TITLE
Fix builds on OS X 10.8

### PR DESCRIPTION
**Pull request by busterb**
This applies a few fixes that allow Mettle to build on OS X 10.8 (what we're targeting binaries on). This also removes a dependency on mbedtls for OSX and Windows.

### CODE REVIEW

**Comment by timwr** [comment]
Great stuff! I can review this, but I'm not in a position to actually test this. I've given it a quick eyeball and it looks reasonable.

File: mettle/configure.ac [hunk 1]
```diff
@@ -19,7 +19,7 @@ case $host_os in
 		HOST_OS=apple
 		OBJCFLAGS="$OBJCFLAGS -fobjc-arc"
 		AC_DEFINE(HAVE_WEBCAM)
-		AC_SUBST([PLATFORM_LDADD], ['-framework CoreMedia -framework CoreImage -framework AVFoundation -framework ImageIO'])
+		AC_SUBST([PLATFORM_LDADD], ['-framework AppKit -framework CoreMedia -framework QuartzCore -framework AVFoundation -framework ImageIO'])
```
Comment (timwr) [inline mettle/configure.ac]: Minor fix: `AppKit` is OSX only so this breaks iOS. https://github.com/bcook-r7/mettle/pull/8 has the fix

**Comment by timwr** [comment]
Looks good to me, added some minor fixes here: https://github.com/bcook-r7/mettle/pull/8

**Comment by busterb** [comment]
We found while testing this that the check for OS X vs iOS was actually backward. Also, the desktop screenshot functionality needs some fixes to build on older releases.

**Comment by timwr** [comment]
Ooops well spotted. I'm not seeing the screenshot functionality both before and after the most recent change. It's also possible I'm doing something stupid or my environment is dirty...

File: mettle/configure.ac [hunk 2]
```diff
@@ -19,17 +19,17 @@ case $host_os in
 		HOST_OS=apple
 		OBJCFLAGS="$OBJCFLAGS -fobjc-arc"
 		AC_DEFINE(HAVE_WEBCAM)
-		AC_SUBST([PLATFORM_LDADD], ['-framework CoreMedia -framework CoreImage -framework AVFoundation -framework ImageIO'])
+		AC_SUBST([PLATFORM_LDADD], ['-framework CoreMedia -framework QuartzCore -framework AVFoundation -framework ImageIO'])
 		AC_EGREP_CPP(yes,
 			[#import <TargetConditionals.h>
-			 #if TARGET_OS_OSX
+			 #ifndef TARGET_OS_OSX
```
Comment (timwr) [inline mettle/configure.ac]: Isn't TARGET_OS_OSX defined to 0 on iOS. so #ifndef is false on both?

File: mettle/configure.ac [hunk 3]
```diff
@@ -19,17 +19,17 @@ case $host_os in
...
 			 yes
 			 #endif],
```
Comment (timwr) [inline mettle/configure.ac]: I wonder if the lexer is treating the  `],` as part of the program
https://github.com/bcook-r7/mettle/pull/9 seems to work?

Comment (busterb) [inline mettle/configure.ac]: Note sure, it should be in config.log at any rate.

For reference, we're building in a boxcutter 10.8 vagrant vm. I can upload it to an S3 bucket if you want to replicate.

File: mettle/src/stdapi/ui/osx_desktop.m [hunk 4]
```diff
@@ -16,7 +16,7 @@
     NSDictionary *properties = [NSDictionary dictionaryWithObjectsAndKeys:
                                 @(compression), kCGImageDestinationLossyCompressionQuality,
                                 nil];
-    CGImageDestinationAddImage(destination, image, (CFDictionaryRef)properties);
+    CGImageDestinationAddImage(destination, image, (__bridge CFDictionaryRef)properties);
```
Comment (busterb) [inline mettle/src/stdapi/ui/osx_desktop.m]: obscure, how did you find this?

Comment (jmartin-r7) [inline mettle/src/stdapi/ui/osx_desktop.m]: The compiler error actually recommended it, looks like an older compiler requirement.

File: mettle/configure.ac (hunk 3 above)
Comment (busterb) [inline mettle/configure.ac]: ok, looks good

**Pull request merged**